examples/enc-search-sample.rs

```diff
@@ -0,0 +1,421 @@
+
+    // Create a group to use for the encryption of customer records.
+    // Note this is not the same as the group that was used to secure
+    // the blind index, but anyone that needs to be able to search for
+    // customers by name will need to belong to both groups.
+    let group_id = create_group_id("customerService")?;
+    create_group(&sdk, &group_id, "Customer Service").await?;
+
+    // Add some test customers to the "server store"
+    let cust1 = encrypt_customer(&sdk, 1, "Gumby", "", &group_id).await?;
+    save_customer(cust1, &[], &[]);
+    let cust2 = encrypt_customer(&sdk, 2, "Æ neid 北亰", "", &group_id).await?;
+    save_customer(cust2, &[], &[]);
+    let cust3 = encrypt_customer(&sdk, 3, "aeneid bei jing", "", &group_id).await?;
+    save_customer(cust3, &[], &[]);
+
+    // Allow the user to enter additional customers
+    let mut next_id = 4;
+    loop {
+        let mut name = String::new();
+        println!("Enter next customer name, or blank line to quit");
+        std::io::stdin()
+            .read_line(&mut name)
+            .expect("error: couldn't read input");
+        name = name.trim().to_string();
+        if &name == "" {
+            break;
+        } else {
+            let mut email = String::new();
+            println!("Enter customer's email");
+            std::io::stdin()
+                .read_line(&mut email)
+                .expect("error: couldn't read input");
+            email = email.trim().to_string();
+
+            let cust = encrypt_customer(&sdk, next_id, &name, &email, &group_id).await?;
+            save_customer(cust, &[], &[]);
+        }
+        next_id += 1;
+    }
+
+    // now create a new customer record to index
+    let mut customer = Customer {
+        id: 23,
+        name: "J. Fred Muggs".to_string(),
+        name_keys: "".to_string(),
+        email: "j.fred.muggs@nowhere.com".to_string(),
+        email_keys: "".to_string(),
+    };
+
+    // Generate the index tokens for the customer name, then encrypt it
+    // start-snippet{indexData}
+    let name_tokens = blind_index
+        .tokenize_data(&customer.name, None)?
+        .into_iter()
+        .collect::<Vec<u32>>();
+    let group_id = create_group_id("customerService")?;
+    let encrypt_opts = DocumentEncryptOpts::with_explicit_grants(
+        None,                  // document ID - create unique
+        None,                  // document name
+        false,                 // don't encrypt to self
+        vec![group_id.into()], // users and groups to which to grant access
+    );
+    let enc_name = sdk
+        .document_encrypt_unmanaged(customer.name.as_bytes(), &encrypt_opts)
+        .await?;
+    // Replace name with encoded encrypted version. Also need to store EDEKs to decrypt name.
+    customer.name = base64::encode(enc_name.encrypted_data());
+    customer.name_keys = base64::encode(enc_name.encrypted_deks());
+    save_customer(customer.clone(), &name_tokens, &[]);
```

Comment:
Why didn't you compute the index for the email and save them as well? You did it for customer 2, but not this one.

Reply:
Restructured the save so it always computes the index